packages/wonder-blocks-modal/components/modal-panel.js

```diff
@@ -88,100 +85,113 @@ export default class ModalPanel extends React.Component<Props> {
             header,
             footer,
             scrollOverflow,
+            showCloseButton,
             color,
             style,
         } = this.props;
 
-        let mainContent =
-            content &&
-            typeof content === "object" &&
-            content.type === ModalContent ? (
-                ((content: any): React.Element<typeof ModalContent>)
-            ) : (
-                <ModalContent>{content}</ModalContent>
-            );
-
-        if (mainContent) {
-            mainContent = React.cloneElement(mainContent, {
-                // Pass the scrollOverflow and header in to the main content
-                scrollOverflow,
-                header: header || mainContent.props.header,
-                // We override the styling of the main content to help position
-                // it if there is a title bar, footer, or close button being
-                // shown. We have to do this here as the ModalContent doesn't
-                // know about things being positioned around it.
-                style: [
-                    !!titleBar && styles.hasTitleBar,
-                    !!footer && styles.hasFooter,
-                    // TODO(jeresig): Replace with <Layout/>
-                    //showCloseButton &&
-                    //    !titleBar &&
-                    //    ((mediaSize) =>
-                    //        mediaSize === "small" &&
-                    //        styles.smallWithCloseButton),
-                    mainContent.props.style,
-                ],
-            });
-        }
-
         return (
-            <View
-                style={[styles.wrapper, color === "dark" && styles.dark, style]}
-            >
-                {this.maybeRenderCloseButton()}
-                {titleBar}
-                {mainContent}
-                {!footer ||
-                (typeof footer === "object" && footer.type === ModalFooter) ? (
-                    footer
-                ) : (
-                    <ModalFooter>{footer}</ModalFooter>
-                )}
-            </View>
+            <MediaLayout styleSheets={styleSheets}>
+                {({styles}) => {
+                    let mainContent =
```

Comment:
I find use of `let` to be a code smell that a function is needed. What do you think about using a function so that this can be `const` instead? Might make the code a little less nested and easier to follow.